### DISCUSSION TITLE
Which django middlewares are required?

### DISCUSSION THREAD

**Discussion by jamesleeht** [vitalik/django-ninja#1271] [Q&A]
When you create a django project it comes with a list of middlewares by default.

If I am not using any django features except an ORM with a simple rest API, do I still need to include all middlewares?

Appreciate any clarificaiton!

**Comment by vitalik** [comment]
This is more of a general django question, but you can follow some of my thoughts here:

if you use django admin UI - OR your authentication uses **cookies**  - bettter keep them all

if not (your auth is like header based and only for clients outiside browser - then these minimal I would always keep

   -  'django.middleware.security.SecurityMiddleware',
   -  'django.middleware.common.CommonMiddleware',
    - 'django.middleware.clickjacking.XFrameOptionsMiddleware',


but be carefull - maybe if future you will add something and it will be painful security issue
